src/devices/steelseries_arctis_7_plus.c

```diff
@@ -91,6 +94,33 @@ static int arctis_7_plus_request_battery(hid_device* device_handle)
     return map(bat, BATTERY_MIN, BATTERY_MAX, 0, 100);
 }
 
+static int arctis_7_plus_send_equalizer_preset(hid_device* device_handle, uint8_t num)
+{
+    // This headset supports only 4 presets:
+    // flat (default), bass boost, smiley, focus
+
+    switch (num) {
+    case 0: {
+        uint8_t flat[MSG_SIZE] = { 0x0, 0x33, 0x18, 0x18, 0x18, 0x18, 0x18, 0x18, 0x18, 0x18, 0x18, 0x18, 0x0 };
+        return hid_write(device_handle, flat, MSG_SIZE);
+    }
+    case 1: {
+        uint8_t bass[MSG_SIZE] = { 0x0, 0x33, 0x1f, 0x20, 0x1a, 0x15, 0x15, 0x16, 0x16, 0x16, 0x16, 0x23, 0x0 };
+        return hid_write(device_handle, bass, MSG_SIZE);
+    }
+    case 2: {
+        uint8_t smiley[MSG_SIZE] = { 0x0, 0x33, 0x1e, 0x1b, 0x15, 0x10, 0x10, 0x13, 0x1b, 0x1e, 0x20, 0x1f, 0x0 };
+        return hid_write(device_handle, smiley, MSG_SIZE);
+    }
+    case 3: {
+        uint8_t focus[MSG_SIZE] = { 0x0, 0x33, 0x0e, 0x16, 0x11, 0x13, 0x20, 0x24, 0x1f, 0x11, 0x18, 0x11, 0x0 };
+        return hid_write(device_handle, focus, MSG_SIZE);
+    }
+    }
+
+    return 0;
```

Comment:
instead of the ending `return 0;` we should use a 
`default:` and then inside write a error message that it needs to be between 0 and 4 and return an HSC_ERROR (maybe a new one, like HSC_OUTOFBOUNDS defined in `device.h`).

Reason we need this redundant is, that if other headsets want also define the `-e` setting, we might modify the check in `main.c` to support more possibilities. So the implementation needs to also check for itself if the parameter is defined.

Reply:
Adjusted according to your recommendation ;)